packages/gatsby-theme-docs/gatsby-node.js

```diff
@@ -233,10 +238,11 @@ exports.createPages = async ({ graphql, actions, reporter }) => {
     actions.createPage({
       // TODO: how should the path be named exactly?
       path: childMdx.fields.slug,
-      component: require.resolve('./src/templates/releases.js'),
+      component: isOverviewPage
+        ? require.resolve('./src/templates/release-notes-list.js')
+        : require.resolve('./src/templates/release-notes-detail.js'),
```

Comment:
After some back and forth I think it's better to just have two separate templates, it makes things easier to implement and separates a bit the concerns, as the pages are indeed different.